### PR TITLE
Appk 887 : integrate Appsflyer

### PR DESCRIPTION
**Pull request by VictorTellez**
## Track all events in AppsFlyer

[JIRA Issue: APPK-887](https://safeboda.atlassian.net/browse/APPK-887)

#### Description
Track all events in AppsFlyer

| Status |
| ------ |
| CLOSES |

#### How to test it manually
Check AppsFlyer dashboard.

### CODE REVIEW

File: app/src/main/java/com/safeboda/commons/analytics/provider/AppsFlyerProvider.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import com.appsflyer.AppsFlyerConversionListener
+import com.appsflyer.AppsFlyerLib
+import com.safeboda.commons.analytics.entity.AnalyticsEvent
+import com.safeboda.commons.analytics.entity.AnalyticsUser
+import com.safeboda.commons.analytics.entity.IS_USER_LOGGED_IN
+
+private const val LOG_TAG = "AppsFlyerProvider"
+
+class AppsFlyerProvider(
+    val app: Application,
```
Comment (Drjacky) [inline app/src/main/java/com/safeboda/commons/analytics/provider/AppsFlyerProvider.kt:15]: Please change it to context, as other providers, google, etc.

File: app/src/main/java/com/safeboda/commons/analytics/provider/AppsFlyerProvider.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    private val appsFlyerInstance = AppsFlyerLib.getInstance()
+        .init(apiKey, conversionDataListener, app.baseContext)
+
+    override fun setUser(user: AnalyticsUser) {
+        // TODO: SET THE .startTracking(app.baseContext)
+        appsFlyerInstance.setCustomerIdAndTrack(user.id.toString(), app.baseContext)
+    }
+
+    override fun clearUser(user: AnalyticsUser) {
+        // appsFlyerInstance.stopTracking(true, app.baseContext)
```
Comment (Drjacky) [inline app/src/main/java/com/safeboda/commons/analytics/provider/AppsFlyerProvider.kt:53]: Let's uncomment this

Comment (VictorTellez) [reply app/src/main/java/com/safeboda/commons/analytics/provider/AppsFlyerProvider.kt:53]: Ok

File: app/src/main/java/com/safeboda/commons/analytics/provider/AppsFlyerAnalyticsProvider.kt [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    }
+
+    private val trackingRequestListener: AppsFlyerTrackingRequestListener = object : AppsFlyerTrackingRequestListener {
+        override fun onTrackingRequestSuccess() {
+            Log.d(LOG_TAG, "trackingListener onTrackingRequestSuccess")
+        }
+
+        override fun onTrackingRequestFailure(error: String?) {
+            Log.d(LOG_TAG, "trackingListener onTrackingRequestFailure :  $error")
```
Comment (Drjacky) [inline app/src/main/java/com/safeboda/commons/analytics/provider/AppsFlyerAnalyticsProvider.kt:52]: Change it to `.e(`

Comment (VictorTellez) [reply app/src/main/java/com/safeboda/commons/analytics/provider/AppsFlyerAnalyticsProvider.kt]: Done

File: dependencies.gradle [hunk 4]
```diff
@@ -1,10 +1,10 @@
 ext {
     // Android
     minSdkVersion = 19
-    targetSdkVersion = 28
-    versionCode = 16
-    versionName = "0.0.16"
-    androidCompileSdkVersion = 28
+    targetSdkVersion = 29
+    versionCode = 17
```
Comment (Drjacky) [inline dependencies.gradle:5]: 18

Comment (VictorTellez) [reply dependencies.gradle:5]: Ok

**Pull request merged**